File: hanul/first_missing_positive.cpp
class Solution {
public:
	int firstMissingPositive(vector<int>& nums) {
		int positiveIdx = 0;

		int length = nums.size();
		for (int i = 0; i < length; ++i)
		{
			if (nums[i] > 0)
			{
				nums[positiveIdx++] = nums[i];
			}
		}

		int max = positiveIdx;
		for (int i = 0; i < positiveIdx; ++i)
		{
			if (abs(nums[i]) - 1 < positiveIdx && nums[abs(nums[i]) - 1] > 0)
			{
				nums[abs(nums[i]) - 1] *= -1;
			}
		}

		for (int i = 0; i < positiveIdx; ++i)
		{
			if (nums[i] > 0)
			{
				return i + 1;
			}
		}

		return positiveIdx + 1;
	}
};

File: Jungkun/First_Missing_Positive.py
# https://leetcode.com/submissions/detail/272506543/

class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        for i in range(len(nums)-1,-1,-1):
            if nums[i]<=0:
                nums.pop(i)
        for i in range(len(nums)):
            if abs(nums[i])<=len(nums) and nums[abs(nums[i])-1]>0 :
                nums[abs(nums[i])-1] *=-1
        for i in range(len(nums)):
            if nums[i]>0:
                return nums.index(nums[i])+1
        return len(nums)+1
